Assets/Scripts/Source/Game/Combat/CombatManager.cs

```diff
@@ -21,14 +37,22 @@ public virtual void Init(CharacterManager characterManager, TurnManager turnMana
 
     public void PlayerAttack()
     {
+        if (!turnManager.ShouldWaitForPlayerAction(characterManager.GetActivePlayer()))
+            return;
+
         Player player = characterManager.GetActivePlayer();
+        player.hitBonus = false;
         turnManager.RegisterAction(player.Attack(characterManager.GetEnemies()));
         onPlayerActionReceived();
     }
 
     public void PlayerDefend()
     {
+        if (!turnManager.ShouldWaitForPlayerAction(characterManager.GetActivePlayer()))
```

Comment:
Should we log a warning here to indicate we're skipping the player's action due to an invalid state?